### PR TITLE
feat: Update eCommerce to send summary events

### PR DESCRIPTION
**Pull request by rmi22186**
## Summary

@alexs-mparticle - The below is quite confusing and we should do a quick chat over zoom. I am summarizing as best I can for posterity. This [Google Sheet](https://docs.google.com/spreadsheets/d/1m_nhSW5_lsxg5k47MX_B9HJPEFeVrQ0UL4QrMLft_c4/edit#gid=1656373099) was created by Amplitude to show the current behavior and the desired behavior.  Given this PR is for web, you should only look at client side behavior.

Currently, our Amplitude ecommerce mappings work as follows using our internal `expandCommerceEvent` method from the core SDK: 
1. **Revenue events:** (ie. a `Purchase` or a `Refund` including 2 products)
a.   expand the event to be 3 events - 1 purchase/refund summary event that uses the amplitude.logRevenueV2 method which results in an `Amplitude Revenue` event, and 2 `eCommerce - purchase - Item` events, one for each product

2. **Non-revenue events** (ie. `AddToCart` including 2 products)
a. expand the event to be 2 items - 2 `eCommerce - add_to_cart - Item` events, one for each product

Amplitude wants us to make a change where we have "summary events", and an option to either include or exclude the individual item events.  Due to backwards compatibility issues with our server team and customer requests, the flag will be called `excludeIndividualProductEvents` (boolean).  All summary events will include a `products` key with a JSON of the products.

We need to support 2 scenarios:
A. scenario A - only a summary event is sent
B. scenario B - To maintain backwards compatibility send a summary event **as well as** the previously sent events, (the item level events and a revenue event if it's a purchase/refund event)

Amplitude's server has special behavior for revenue events, and reads keys prefixed with `$`.  These are called reserved attributes and help Amplitude calculate things like customer LTV.   Since we need to both be forward looking but backwards compatible, there is additional logic for revenue events.  Non revenue events are not impacted by the `$` prefix:
*  For Amplitude revenue events, when `excludeIndividualProductEvents` = true, we send a single summary event and need to prefix `price` and `revenue` with a `$` so that Amplitude knows to use these values.
* * When `excludeIndividualProductEvents` = false, we support Scenario B, and we need to remove the `$` from `price` and `revenue` from the summary event, and send the traditional revenue event via the `logRevenueV2`, which will automatically append the `$`. We avoid manually appending `$` in the summary event in this case to avoid Amplitude double counting.

Finally, Amplitude requests that we append a key `mparticle_amplitude_should_split` so their servers know that this update has been made.

## Testing Plan
Updated ecommerce unit tests which failed due to the changing ecommerce mappings.  Added tests for revenue and non revenue events for when the flag `excludeIndividualProductEvents` is `true`, and `false`.

## Reference Issue
- Closes https://go.mparticle.com/work/SQDSDKS-4388

### CODE REVIEW

**Review by alexs-mparticle — CHANGES_REQUESTED** [review]

File: src/Amplitude.js [hunk 1]
```diff
@@ -328,7 +337,7 @@ var constructor = function () {
     function createEcommerceAttributes(attributes) {
         var updatedAttributes = {};
         for (var key in attributes) {
-            if (key !== 'Total Amount' && key !== 'Total Product Amount') {
+            if (key !== TOTAL_AMOUNT && key !== 'Total Product Amount') {
```
Comment (alexs-mparticle) [inline src/Amplitude.js:340]: You missed `TOTAL_PRODUCT_AMOUNT`

File: src/Amplitude.js [hunk 2]
```diff
@@ -338,62 +347,203 @@ var constructor = function () {
...
-            var isRefund, isPurchase, logRevenue;
-            isRefund =
-                event.ProductAction.ProductActionType ===
-                mParticle.ProductActionType.Refund;
-            isPurchase =
-                event.ProductAction.ProductActionType ===
-                mParticle.ProductActionType.Purchase;
-            logRevenue = isRefund || isPurchase;
+            var wasProcessed = processProductAction(event, expandedEvents);
+            return wasProcessed;
```
Comment (alexs-mparticle) [inline src/Amplitude.js:353]:
```suggestion
            return processProductAction(event, expandedEvents);
```

File: src/Amplitude.js [hunk 3]
```diff
@@ -338,62 +347,203 @@ var constructor = function () {
...
+                    expandedEvt.EventName,
+                    updatedAttributes
+                );
             });
 
             return true;
         }
 
+        console.warn(
+            'Commerce event does not conform to our expectations and was not forwarded to Amplitude. Please double check your code.'
```
Comment (alexs-mparticle) [inline src/Amplitude.js:374]:
```suggestion
            'Commerce event does not conform to our expectations and was not forwarded to Amplitude. Please double-check your code.'
```

File: src/Amplitude.js [hunk 4]
```diff
@@ -506,7 +662,7 @@ var constructor = function () {
     this.setUserAttribute = setUserAttribute;
     this.setOptOut = setOptOut;
     this.removeUserAttribute = removeUserAttribute;
-};
+};;
```
Comment (alexs-mparticle) [inline src/Amplitude.js:665]:
```suggestion
};
```

**Review by alexs-mparticle — CHANGES_REQUESTED** [review]

File: src/Amplitude.js (hunk 3 above)
Comment (rmi22186) [reply src/Amplitude.js:374]: TIL there's a hyphen.

File: src/Amplitude.js [hunk 5]
```diff
@@ -338,62 +347,203 @@ var constructor = function () {
...
         }
 
+        console.warn(
+            'Commerce event does not conform to our expectations and was not forwarded to Amplitude. Please double check your code.'
+        );
+
+        return false;
+    }
+
+    function processProductAction(unexpandedCommerceEvent, expandedEvents) {
```
[the remark this answers is not in the document]
Comment (rmi22186) [reply src/Amplitude.js:379]: i added a .MD file in the test folder to show this.  I think as a comment might be a little too big.  so i referenced it there too.

File: test/AmplitudeCommerceEvents.MD [hunk 6]
```diff
@@ -0,0 +1,87 @@
+## Revenue Commerce Events (purchases and refunds)
```
Comment (alexs-mparticle) [inline test/AmplitudeCommerceEvents.MD:1]: I think this file should be under a `/docs` folder

**Review by alexs-mparticle — APPROVED** [review]

**Pull request merged**